src/transformers/models/cvt/modeling_cvt.py

```diff
@@ -0,0 +1,750 @@
+# limitations under the License.
+""" PyTorch CvT model. """
+
+
+import collections.abc
+from collections import OrderedDict
+
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+import torch.utils.checkpoint
+from torch.nn import CrossEntropyLoss, MSELoss
+
+from einops import rearrange
+from einops.layers.torch import Rearrange
+from timm.models.layers import DropPath, trunc_normal_
+
+from ...file_utils import add_start_docstrings, add_start_docstrings_to_model_forward, replace_return_docstrings
+from ...modeling_outputs import BaseModelOutputWithPooling, SequenceClassifierOutput
+from ...modeling_utils import PreTrainedModel
+from ...utils import logging
+from .configuration_cvt import CvTConfig
+
+
+logger = logging.get_logger(__name__)
+
+_CONFIG_FOR_DOC = "CvTConfig"
+_CHECKPOINT_FOR_DOC = "microsoft/cvt-21-384x384-IN-1k"
+
+CVT_PRETRAINED_MODEL_ARCHIVE_LIST = [
+    "microsoft/cvt-21-384x384-IN-1k",
+    # See all CvT models at https://huggingface.co/models?filter=CvT
+]
+
+
+# see for official implementation by
+# https://github.com/microsoft/CvT
+
+
+def to_2tuple(x):
+    if isinstance(x, collections.abc.Iterable):
+        return x
+    return (x, x)
+
+
+class PatchEmbeddings(nn.Module):
+    """Image to Conv Embedding"""
+
+    def __init__(self, patch_size=7, in_chans=3, embed_dim=64, stride=4, padding=2, norm_layer=None):
+        super().__init__()
+        patch_size = to_2tuple(patch_size)
+        self.patch_size = patch_size
+
+        self.proj = nn.Conv2d(in_chans, embed_dim, kernel_size=patch_size, stride=stride, padding=padding)
+        self.norm = norm_layer(embed_dim) if norm_layer else None
+
+    def forward(self, x):
+        x = self.proj(x)
+        B, C, H, W = x.shape
+        x = rearrange(x, "b c h w -> b (h w) c")
+        if self.norm:
+            x = self.norm(x)
+        x = rearrange(x, "b (h w) c -> b c h w", h=H, w=W)
+        return x
+
+
+class CvTOutput(nn.Module):
+    def __init__(self, in_features, hidden_features=None, out_features=None, act_layer=nn.GELU, drop=0.0):
+        super().__init__()
+        out_features = out_features or in_features
+        hidden_features = hidden_features or in_features
+        self.fc1 = nn.Linear(in_features, hidden_features)
+        self.fc2 = nn.Linear(hidden_features, out_features)
+        self.act = act_layer()
+        self.drop = nn.Dropout(drop)
+
+    def forward(self, x):
+        x = self.fc1(x)
+        x = self.act(x)
+        x = self.drop(x)
+        x = self.fc2(x)
+        x = self.drop(x)
+        return x
+
+
+class CvTAttention(nn.Module):
+    def __init__(
+        self,
+        dim_in,
+        dim_out,
+        num_heads,
+        qkv_bias=False,
+        attn_drop=0.0,
+        proj_drop=0.0,
+        method="dw_bn",
+        kernel_size=3,
+        stride_kv=1,
+        stride_q=1,
+        padding_kv=1,
+        padding_q=1,
+        with_cls_token=True,
+        **kwargs
+    ):
```

Comment:
```suggestion
class CvTAttention(nn.Module):
    def __init__(self, config):
```
It would be great if this class can only take config as input.